diff --git a/panel-plugin/devices.c b/panel-plugin/devices.c
index 47c640e..797b079 100644
--- a/panel-plugin/devices.c
+++ b/panel-plugin/devices.c
@@ -180,14 +180,14 @@ disk_print (t_disk * pdisk)
 }
 
 char *
-shorten_disk_name (const char *dev, gint len)
+shorten_disk_name (const char *dev, guint len)
 {
     char *r, *lastchars, *firstchars;
     if (strncmp(dev, "LABEL=", 6)==0)
     {
       r = g_strdup(dev+6*sizeof(char));
     }
-    else if (strlen(dev)>len) // len canot be set lower than 9
+    else if (strlen(dev)>len) // len cannot be set lower than 9
     {
         // we want at least 5 characters at the end so that trimmed UUIDs are still readable
         lastchars = (char *) (dev + strlen(dev) - 5);
@@ -346,7 +346,7 @@ disk_umount (t_disk *pdisk, char* umount_command, gboolean show_message_dialog,
     {
 
         DBG("disk_umount: dev=%s, mountpoint=%s, umount_command=%s, show_message_dialog=%d, eject=%d, type=%s", pdisk->device, pdisk->mount_point, umount_command, show_message_dialog, eject, pdisk->mount_info->type);
-        if (strstr(pdisk->mount_info->type, "fuse"))
+        if (strstr(pdisk->mount_info->type, "fuse."))
           deviceprintf(&tmp, "fusermount -u %m", pdisk->device);
         else
           deviceprintf(&tmp, umount_command, pdisk->device);
@@ -418,7 +418,7 @@ gboolean
 device_or_mountpoint_exists (GPtrArray * pdisks, t_disk * pdisk)
 {
   gboolean returnValue = FALSE;
-  int i;
+  guint i;
   t_disk *disk;
   int stringlength1, stringlength2, stringlength3, stringlength4;
 
diff --git a/panel-plugin/devices.h b/panel-plugin/devices.h
index 2dd65bc..fb87705 100644
--- a/panel-plugin/devices.h
+++ b/panel-plugin/devices.h
@@ -107,15 +107,17 @@ void disk_mount (t_disk *pdisk, char *on_mount_cmd, char* mount_command, gboolea
  */
 void disk_umount (t_disk *pdisk, char* umount_command, gboolean show_message_dialog, gboolean eject);
 
+gboolean device_or_mountpoint_exists (GPtrArray * pdisks, t_disk * pdisk);
 
 /**
  * Fill a GPtrArray with pointers on struct t_disk containing infos on devices
  * and theoretical mount point. Uses setfsent() and getfsent().
  * @param include_NFSs    TRUE if remote file systems should also be displayed.
  * @param showed_fstab_dialog Pointer to whether dialog has already been shown before
+ * @param length Length of the mount point names to display when shortened
  * @return                GPtrArray with t_disks
  */
-GPtrArray * disks_new (gboolean include_NFSs, gboolean *showed_fstab_dialog, gint length) ;
+GPtrArray * disks_new (gboolean include_NFSs, gboolean *showed_fstab_dialog, gint length);
 
 
 /**
@@ -195,9 +197,9 @@ void mount_info_print(t_mount_info * mount_info);
 t_mount_info * mount_info_new (float size, float used, float avail, unsigned int percent, char * type, char * mounted_on);
 void mount_info_free(t_mount_info * * mount_info);
 void disk_print (t_disk * pdisk);
-char * shorten_disk_name (const char *dev, gint length);
+char * shorten_disk_name (const char *dev, guint length);
 t_disk * disk_new (const char * dev, const char * mountpoint, gint length);
 void disk_free(t_disk **pdisk);
 void disks_free_mount_info(GPtrArray * pdisks);
 
 #endif /* _DEVICES_H_ */
