--- a/panel-plugin/button.h
+++ b/panel-plugin/button.h
@@ -26,6 +26,7 @@
 #define _XFCE_PANEL_PLACES_BUTTON_H
 
 #include <gtk/gtk.h>
+#include <libxfce4panel/libxfce4panel.h>
 
 #define PLACES_TYPE_BUTTON             (places_button_get_type ())
 #define PLACES_BUTTON(obj)             (G_TYPE_CHECK_INSTANCE_CAST ((obj), PLACES_TYPE_BUTTON, PlacesButton))
@@ -62,10 +63,10 @@
 };
 
 GType
-places_button_get_type();
+places_button_get_type(void);
 
 GtkWidget*
-places_button_new();
+places_button_new(XfcePanelPlugin *plugin);
 
 void
 places_button_set_label(PlacesButton*, const gchar *label);
--- a/panel-plugin/model.c
+++ b/panel-plugin/model.c
@@ -29,7 +29,7 @@
 
 /********** PlacesBookmarkAction **********/
 
-inline PlacesBookmarkAction*
+PlacesBookmarkAction*
 places_bookmark_action_create(gchar *label)
 {
     PlacesBookmarkAction *action;
@@ -40,7 +40,7 @@
     return action;
 }
 
-inline void
+void
 places_bookmark_action_destroy(PlacesBookmarkAction *act)
 {
     g_assert(act != NULL);
@@ -51,7 +51,7 @@
     g_free(act);
 }
 
-inline void
+void
 places_bookmark_action_call(PlacesBookmarkAction *act)
 {
     g_assert(act != NULL);
@@ -66,7 +66,7 @@
 static int bookmarks = 0;
 #endif
 
-inline PlacesBookmark*
+PlacesBookmark*
 places_bookmark_create(gchar *label)
 {
     PlacesBookmark *bookmark;
@@ -92,7 +92,7 @@
     g_list_free(actions);
 }
 
-inline void
+void
 places_bookmark_destroy(PlacesBookmark *bookmark)
 {
     g_assert(bookmark != NULL);
@@ -121,7 +121,7 @@
 
 /********** PlacesBookmarkGroup **********/
 
-inline GList*
+GList*
 places_bookmark_group_get_bookmarks(PlacesBookmarkGroup *pbg)
 {
     g_assert(pbg->get_bookmarks != NULL);
@@ -129,7 +129,7 @@
     return pbg->get_bookmarks(pbg);
 }
 
-inline gboolean
+gboolean
 places_bookmark_group_changed(PlacesBookmarkGroup *pbg)
 {
     g_assert(pbg->changed != NULL);
@@ -137,7 +137,7 @@
     return pbg->changed(pbg);
 }
 
-inline PlacesBookmarkGroup*
+PlacesBookmarkGroup*
 places_bookmark_group_create(void)
 {
     PlacesBookmarkGroup *bookmark_group;
@@ -146,7 +146,7 @@
     return bookmark_group;
 }
 
-inline void
+void
 places_bookmark_group_destroy(PlacesBookmarkGroup *pbg)
 {
     if(pbg->finalize != NULL)
--- a/panel-plugin/model.h
+++ b/panel-plugin/model.h
@@ -36,13 +36,13 @@
     void        (*finalize) (PlacesBookmarkAction *self);
 };
 
-inline PlacesBookmarkAction*
+PlacesBookmarkAction*
 places_bookmark_action_create(gchar *label);
 
-inline void
+void
 places_bookmark_action_destroy(PlacesBookmarkAction*);
 
-inline void
+void
 places_bookmark_action_call(PlacesBookmarkAction*);
 
 /* Places Bookmark */
@@ -70,10 +70,10 @@
     void                 (*finalize) (PlacesBookmark *self);
 };
 
-inline PlacesBookmark*
+PlacesBookmark*
 places_bookmark_create(gchar *label);
 
-inline void
+void
 places_bookmark_destroy(PlacesBookmark *bookmark);
 
 /* Places Bookmark Group */
@@ -86,16 +86,16 @@
     gpointer    priv;
 };
 
-inline GList*
+GList*
 places_bookmark_group_get_bookmarks(PlacesBookmarkGroup*);
 
-inline gboolean
+gboolean
 places_bookmark_group_changed(PlacesBookmarkGroup*);
 
-inline PlacesBookmarkGroup*
-places_bookmark_group_create();
+PlacesBookmarkGroup*
+places_bookmark_group_create(void);
 
-inline void
+void
 places_bookmark_group_destroy(PlacesBookmarkGroup*);
 
 #endif
--- a/panel-plugin/model_system.h
+++ b/panel-plugin/model_system.h
@@ -25,7 +25,7 @@
 #include "model.h"
 
 PlacesBookmarkGroup*
-places_bookmarks_system_create();
+places_bookmarks_system_create(void);
 
 #endif
 /* vim: set ai et tabstop=4: */
--- a/panel-plugin/model_user.h
+++ b/panel-plugin/model_user.h
@@ -25,7 +25,7 @@
 #include "model.h"
 
 PlacesBookmarkGroup*
-places_bookmarks_user_create();
+places_bookmarks_user_create(void);
 
 #endif
 /* vim: set ai et tabstop=4: */
--- a/panel-plugin/model_volumes.h
+++ b/panel-plugin/model_volumes.h
@@ -25,7 +25,7 @@
 #include "model.h"
 
 PlacesBookmarkGroup*
-places_bookmarks_volumes_create();
+places_bookmarks_volumes_create(gboolean mount_and_open_by_default);
 
 #endif
 /* vim: set ai et tabstop=4: */
