--- a/src/xfcetimer.c
+++ b/src/xfcetimer.c
@@ -1150,7 +1150,7 @@
 
 
           groupnum=0;
-          g_sprintf(groupname,"G0");
+          g_snprintf(groupname, sizeof groupname, "G0");
 
 
           while(xfce_rc_has_group(rc,groupname)){
@@ -1179,7 +1179,7 @@
 			 alrm->pd = (gpointer) pd;
 			 
              groupnum++;
-             g_snprintf(groupname,5,"G%d",groupnum);
+             g_snprintf(groupname, sizeof groupname, "G%d", groupnum);
 
           } /* end of while loop */
 
