From 60d8fc24b66c93d29d25c9112b08865e57e26cc5 Mon Sep 17 00:00:00 2001
From: Akbarkhon Variskhanov <akbarkhon.variskhanov@gmail.com>
Date: Sun, 4 Dec 2022 14:33:57 +0500
Subject: [PATCH] Fix strings if song data is not available (#9)

---
 panel-plugin/xfce4-mpc-plugin.c | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/panel-plugin/xfce4-mpc-plugin.c b/panel-plugin/xfce4-mpc-plugin.c
index 1149e4f..87e2054 100644
--- a/panel-plugin/xfce4-mpc-plugin.c
+++ b/panel-plugin/xfce4-mpc-plugin.c
@@ -439,13 +439,27 @@
    if (0 == str->len)
       g_string_assign(str, mpc->playlist_format);
 
-   /* replace %artist% by song->artist, etc */
+   /* replace format strings with song data or placeholders */
+   if (song->artist == NULL)
+      song->artist = g_strdup(_("Unknown Artist"));
    str_replace(str, "%artist%", song->artist);
+
+   if (song->album == NULL)
+      song->album = g_strdup(_("Unknown Album"));
    str_replace(str, "%album%", song->album);
+
+   if (song->title == NULL)
+      song->title = g_strdup(_("Unknown Title"));
    str_replace(str, "%title%", song->title);
+
+   if (song->track == NULL)
+      song->track = g_strdup(_("Unknown Track"));
    str_replace(str, "%track%", song->track);
-}
 
+   if (song->file == NULL)
+      song->file = g_strdup(_("Unknown File"));
+   str_replace(str, "%file%", song->file);
+}
 static void
 enter_cb(GtkWidget *widget, GdkEventCrossing* event, t_mpc* mpc)
 {
