--- a/scripts/xinitrc.in.in
+++ b/scripts/xinitrc.in.in
@@ -33,6 +33,7 @@
 # or empty, a default equal to $HOME/.config should be used.
 if test "x$XDG_CONFIG_HOME" = "x" ; then
   XDG_CONFIG_HOME=$HOME/.config
+  export XDG_CONFIG_HOME
 fi
 [ -d "$XDG_CONFIG_HOME" ] || mkdir "$XDG_CONFIG_HOME"
 
@@ -41,6 +42,7 @@
 # set or empty, a default equal to $HOME/.cache should be used.
 if test "x$XDG_CACHE_HOME" = "x" ; then
   XDG_CACHE_HOME=$HOME/.cache
+  export XDG_CACHE_HOME
 fi
 [ -d "$XDG_CACHE_HOME" ] || mkdir "$XDG_CACHE_HOME"
 
@@ -87,6 +89,35 @@
 # load local modmap
 test -r $HOME/.Xmodmap && xmodmap $HOME/.Xmodmap
 
+# ensure both the DBus session bus and systemd user session (if running) has
+# all the env vars it needs to properly populate the environment of child
+# processes
+if command -v dbus-update-activation-environment >/dev/null 2>&1; then
+  if command -v systemctl >/dev/null 2>&1 && systemctl --user list-jobs >/dev/null 2>&1; then # user session is running
+    systemd_arg='--systemd'
+  fi
+
+  dbus-update-activation-environment $systemd_arg \
+    DESKTOP_SESSION \
+    XAUTHLOCALHOSTNAME=$XAUTHLOCALHOSTNAME \
+    XDG_CACHE_HOME \
+    XDG_CONFIG_DIRS \
+    XDG_CONFIG_HOME \
+    XDG_CURRENT_DESKTOP \
+    XDG_DATA_DIRS \
+    XDG_DATA_HOME \
+    XDG_MENU_PREFIX \
+    XDG_RUNTIME_DIR \
+    XDG_SEAT \
+    XDG_SEAT_PATH \
+    XDG_SESSION_CLASS \
+    XDG_SESSION_DESKTOP \
+    XDG_SESSION_ID \
+    XDG_SESSION_PATH \
+    XDG_SESSION_TYPE \
+    XDG_STATE_HOME
+fi
+
 # run xfce4-session if installed
 if which xfce4-session >/dev/null 2>&1; then
 
