From f2d079da91e292cb98ad3ac10beece931d7fca79 Mon Sep 17 00:00:00 2001
From: Andre Miranda <andre42m@gmail.com>
Date: Fri, 4 Nov 2016 22:54:27 -0300
Subject: [PATCH 1/6] Fix compilation warn

---
 panel-plugin/xfce4-cpufreq-plugin.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/panel-plugin/xfce4-cpufreq-plugin.h b/panel-plugin/xfce4-cpufreq-plugin.h
index e610f53..30308d1 100644
--- a/panel-plugin/xfce4-cpufreq-plugin.h
+++ b/panel-plugin/xfce4-cpufreq-plugin.h
@@ -109,6 +109,9 @@ cpufreq_restart_timeout (void);
 void
 cpufreq_write_config (XfcePanelPlugin *plugin);
 
+void
+cpufreq_update_icon (CpuFreqPlugin *cpufreq);
+
 G_END_DECLS
 
 #endif /* XFCE4_CPU_FREQ_H */
-- 
2.45.2

